api/src/test/java/org/openmrs/module/bedmanagement/BedManagementDaoImplTest.java

```diff
@@ -0,0 +1,175 @@
+package org.openmrs.module.bedmanagement;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterType;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.bedmanagement.dao.BedManagementDao;
+import org.openmrs.module.bedmanagement.entity.Bed;
+import org.openmrs.module.bedmanagement.entity.BedPatientAssignment;
+import org.openmrs.module.bedmanagement.entity.BedType;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class BedManagementDaoImplTest extends BaseModuleContextSensitiveTest {
+	
+	private BedManagementDao bedManagementDao;
+	
+	private EncounterType defaultEncounterType;
+	
+	@Before
+	public void setup() throws Exception {
+		executeDataSet("org/openmrs/include/standardTestDataset.xml");
+		
+		bedManagementDao = Context.getRegisteredComponent("bedManagementDao", BedManagementDao.class);
+		
+		defaultEncounterType = Context.getEncounterService().getEncounterType("Scheduled");
+	}
+	
+	private Bed createBed(String bedNumber) {
+		BedType bedType = new BedType();
+		bedType.setName("Test BedType");
+		bedType.setDisplayName("TB");
+		bedType.setDescription("Test Bed Type");
+		bedType = bedManagementDao.saveBedType(bedType);
+		
+		Bed bed = new Bed();
+		bed.setBedNumber(bedNumber);
+		bed.setStatus("AVAILABLE");
+		bed.setBedType(bedType);
+		
+		return bedManagementDao.saveBed(bed);
+	}
+	
+	private Visit getTestVisit(Patient patient) {
+		return Context.getVisitService().getVisitsByPatient(patient).get(0);
```

Comment:
You can make this test more efficient by directly fetching this encounter with VisitService.getVisit(id)